OneCommunityGlobal/HGNRest
# Angela - Pinterest Poster

# Description
This PR adds API endpoints for posting content on Pinterest, enabling image extraction and posting capabilities (note: scheduling is not included in this PR). Currently, there are no related frontend PRs, as frontend development for these features is still in progress. Once the frontend is ready, these APIs will be integrated to complete the functionality.

## Main changes explained:
- Created `socialMediaController.js`: 
	•	Extract text and image URLs or base64 content from content.
	•	Post images to Pinterest using createPin.
	•	Handle token retrieval for Pinterest OAuth authentication.
- Created `socialMediaRouter.js`: Defines routes to handle Pinterest API interactions, including posting images and obtaining access tokens.
- Updated `routes.js`: Integrated the new socialMediaRouter within the main app router configuration to expose the new API endpoints.

## How to test:
1. check into current branch
2. do `npm install` and `npm run dev` to run this PR locally
3. Access the endpoints:
- /api/getPinterestAccessToken - Test Pinterest token retrieval.
- /api/createPin - Test Pinterest posting functionality with sample data.

## Screenshots or videos of changes (Just temporary UI):
https://www.loom.com/share/e41dadeae7334dc9a20a1adff8b4b3fc?sid=ef77017e-d178-44b7-ba45-9d2105576708

## Note:
1. Pinterest Business Account:
- Ref: https://developers.pinterest.com/docs/getting-started/connect-app/
- For developing and testing, trial access is enough.
- For real world usage, you have to apply for standard access.

2. Set them as environment variables in your application (.env file):

Pinterest:
- REACT_APP_PINTEREST_CLIENT_ID
- REACT_APP_PINTEREST_CLIENT_SECRET
